semeio/jobs/design_kw/design_kw.py

```diff
@@ -116,3 +119,53 @@ def extract_key_value(parameters):
     if errors:
         raise ValueError("\n".join(errors))
     return res
+
+
+def rm_genkw_prefix(str_or_dict, ignoreprefixes="LOG10_"):
```

Comment:
Unless I've misunderstood the type of `str_or_dict` will always be a dict on the first call, and then the string support is used internally on the keys afterwards. I think the implementation would be much cleaner if we utilised this assumption and hid the string-support within the function such that `rm_genkw_prefix` only accepts dicts of depth one.

Reply:
New  implementation written - it got a lot better with only dict as input.